sdk/storage/Azure.Storage.Files/src/Generated/FileRestClient.cs

```diff
@@ -6937,6 +6937,96 @@ internal static Azure.Storage.Files.Models.FileProperty FromXml(System.Xml.Linq.
 }
 #endregion class FileProperty
 
+#region struct FileRange
+namespace Azure.Storage.Files.Models
+{
+    /// <summary>
+    /// An Azure Storage file range.
+    /// </summary>
+    public readonly partial struct FileRange: System.IEquatable<FileRange>
+    {
+        /// <summary>
+        /// Start of the range.
+        /// </summary>
+        public long Start { get; }
+
+        /// <summary>
+        /// End of the range.
+        /// </summary>
+        public long End { get; }
+
+        /// <summary>
+        /// Prevent direct instantiation of FileRange instances.
+        /// You can use FilesModelFactory.FileRange instead.
+        /// </summary>
+        internal FileRange(
+            long start,
+            long end)
+            {
+                Start = start;
+                End = end;
+            }
+
+        /// <summary>
+        /// Check if two FileRange instances are equal.
+        /// </summary>
+        /// <param name="other">The instance to compare to.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        [System.ComponentModel.EditorBrowsable((System.ComponentModel.EditorBrowsableState.Never))]
```

Comment:
I don't think we should hide this overload.

Reply:
Please file a bug against the generator and we'll get to this after the current push.